configTools/src/main/scala/com/gu/janus/policy/Iam.scala

```diff
@@ -0,0 +1,147 @@
+package com.gu.janus.policy
+
+import io.circe.syntax._
+import io.circe.{Encoder, Json, JsonObject}
+
+import scala.collection.immutable.ListMap
+
+/** This models the AWS IAM policy types and subtypes.
+  *
+  * See https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies.html
+  * for the logic of the Json encoding.
+  */
+object Iam {
+
+  sealed trait Effect
+  object Effect {
+    case object Allow extends Effect
+    case object Deny extends Effect
+
+    implicit val encoder: Encoder[Effect] = Encoder.instance {
+      case Allow => Json.fromString("Allow")
+      case Deny  => Json.fromString("Deny")
+    }
+  }
+
+  case class Action(name: String)
+  object Action {
+    implicit val encoder: Encoder[Action] = Encoder[String].contramap(_.name)
+  }
+
+  case class Resource(id: String)
+  object Resource {
+    implicit val encoder: Encoder[Resource] = Encoder[String].contramap(_.id)
+  }
+
+  case class Condition(
+      key: String,
+      typeName: String,
+      conditionValues: Seq[String]
+  )
+  object Condition {
+    implicit val encoder: Encoder[Condition] = Encoder.instance { condition =>
+      Json.obj(
+        condition.typeName -> Json.obj(
+          condition.key -> Json.fromValues(
+            condition.conditionValues.map(Json.fromString)
+          )
+        )
+      )
+    }
+  }
+
+  case class Principal(id: String, provider: String)
+  object Principal {
+    implicit val encoder: Encoder[Principal] = Encoder.instance { principal =>
+      Json.obj(
+        principal.provider -> Json.fromString(principal.id)
+      )
+    }
+  }
+
+  case class Statement(
+      effect: Effect,
+      actions: Seq[Action],
+      resources: Seq[Resource],
+      id: Option[String] = None,
+      conditions: Seq[Condition] = Nil,
+      principals: Seq[Principal] = Nil
+  )
+  object Statement {
+    private def conditionsAsJson(conditions: Seq[Condition]): Json = {
+      val mergedConditions = conditions
+        .groupBy(_.typeName)
+        .view
+        .mapValues(conditions =>
+          Json.obj(
```

Comment:
Would it be clearer to use `Json.fromFields` rather than passing the collection as varargs (`: _*`) to the `Json.obj` constructor?

Reply:
Done in ab1e448ab285a1b31c1b9a57ccee15420e4f99ee